mesonpy/_editable.py

```diff
@@ -304,12 +304,14 @@ def rebuild(self) -> Node:
         env[MARKER] = os.pathsep.join((env.get(MARKER, ''), self._build_path))
 
         if self._verbose or bool(env.get(VERBOSE, '')):
-            print('+ ' + ' '.join(self._build_cmd))
-            stdout = None
+            dry_run_build_cmd = self._build_cmd + ['-n']
+            # We do not want any output if there is no work to do
+            p = subprocess.run(dry_run_build_cmd, cwd=self._build_path, env=env, capture_output=True)
+            if b'no work to do' not in p.stdout:
```

Comment:
Relevant: https://github.com/mesonbuild/meson/blob/a35d4d368a21f4b70afa3195da4d6292a649cb4c/mesonbuild/mtest.py#L1636

Reply:
already tested code, always nice:)

Reply:
Probably worth taking that over verbatim, and adding a code comment showing its origin so that if we need to make further tweaks, we can keep the two pieces of code in sync?

Reply:
That would be my recommendation, yes.